src/Raven.Server/Documents/PeriodicBackup/Retention/RetentionPolicyRunnerBase.cs

```diff
@@ -56,23 +57,25 @@ public async Task Execute()
                 var foldersToDelete = new List<string>();
                 var now = DateTime.Now; // the time in the backup folder name is the local time
                 var deletingAllFolders = true;
+                var hasMore = true;
 
-                do
+                while (hasMore)
                 {
-                    var folders = await GetSortedFolders();
-                    _onProgress.Invoke($"Got {folders.List.Count:#,#} potential backups, has more: {folders.HasMore}");
+                    var foldersResult = await GetSortedFolders();
+
+                    _onProgress.Invoke($"Got {foldersResult.List.Count:#,#} potential backups, has more: {foldersResult.HasMore}");
```

Comment:
This is visible to the user, no? In this case, let's make this more user friendly message

Reply:
done

Reply:
done